server/services/message-service.js

```diff
@@ -1,69 +1,52 @@
 const auth = require('@azure/ms-rest-nodeauth')
-const MessageSender = require('./messaging/message-sender')
-const MessageReceiver = require('./messaging/message-receiver')
-const { claimMessageAction } = require('./message-action')
 const config = require('../config')
+const { claimMessageAction } = require('./message-action')
+const MessageReceiver = require('./messaging/message-receiver')
+const MessageSender = require('./messaging/message-sender')
 
-let calculationSender
-let scheduleSender
-let claimReceiver
-
-async function createConnections () {
-  const credentials = config.isProd ? await auth.loginWithVmMSI({ resource: 'https://servicebus.azure.net' }) : undefined
-  calculationSender = new MessageSender('claim-service-calculation-sender', config.messageQueues.calculationQueue, credentials)
-  scheduleSender = new MessageSender('claim-service-schedule-sender', config.messageQueues.scheduleQueue, credentials)
-  claimReceiver = new MessageReceiver('claim-service-claim-receiver', config.messageQueues.claimQueue, credentials)
-}
-
-async function registerQueues () {
-  await claimReceiver.setupReceiver(claim => {
-    claimMessageAction(claim, publishClaim)
-  })
-}
-
-async function closeConnections () {
-  await calculationSender.closeConnection()
-  await scheduleSender.closeConnection()
-  await claimReceiver.closeConnection()
-}
+class MessageService {
+  constructor (credentials) {
+    this.publishClaim = this.publishClaim.bind(this)
+    this.closeConnections = this.closeConnections.bind(this)
+    this.calculationSender = new MessageSender('claim-service-calculation-sender', config.messageQueues.calculationQueue, credentials)
+    this.scheduleSender = new MessageSender('claim-service-schedule-sender', config.messageQueues.scheduleQueue, credentials)
+    this.claimReceiver = new MessageReceiver('claim-service-claim-receiver', config.messageQueues.claimQueue, credentials)
+  }
 
-async function publishClaim (claim) {
-  try {
-    await Promise.all([
-      calculationSender.sendMessage(claim),
-      scheduleSender.sendMessage(claim)
-    ])
-  } catch (err) {
-    console.log(err)
-    throw err
+  async closeConnections () {
+    await this.calculationSender.closeConnection()
+    await this.scheduleSender.closeConnection()
+    await this.claimReceiver.closeConnection()
   }
-}
 
-function getCalculationSender () {
-  return calculationSender
-}
+  async publishClaim (claim) {
+    try {
+      await Promise.all([
+        this.calculationSender.sendMessage(claim),
+        this.scheduleSender.sendMessage(claim)
+      ])
+    } catch (err) {
+      console.log(err)
+      throw err
+    }
+  }
 
-function getScheduleSender () {
-  return scheduleSender
+  getScheduleMessage (claim) {
```

Comment:
This is an odd function and is only used in `test/contract/schedule.test.js`. I'm not sure it should be here but didn't want to add extra changes.

Reply:
Assuming from the test location it's something to do with the contract testing.